### PR TITLE
Bug 1404010 - Extend the worker to have the runId of its most recentTask

### PR DESCRIPTION
**Pull request by helfi92**
Use `Entity.types.JSON` with `[{taskId, runId}, ...]` instead of `Entity.types.SlugIdArray`.

### CODE REVIEW

File: src/data.js [hunk 1]
```diff
@@ -892,6 +892,32 @@ let Worker = Entity.configure({
...
+    recentTasks:      Entity.types.JSON,
+    disabled:         Entity.types.Boolean,
+    // the time at which this worker should no longer be displayed
+    expires:          Entity.types.Date,
+    firstClaim:       Entity.types.Date,
+  },
+  migrate(item) {
+    item.firstClaim = new Date(2000, 0, 1);
+    item.recentTasks = [];
+    item.disabled = false;
```
Comment (djmitche) [inline src/data.js]: Only `recentChanges` needs to be updated here -- the other two fields have not changed.

File: src/workerinfo.js [hunk 2]
```diff
@@ -183,13 +182,14 @@ class WorkerInfo {
...
     }
 
-    const recentTasks = worker.recentTasks.clone();
+    const recentTasks = [...worker.recentTasks];
 
-    tasks.forEach((task, index) => {
-      const taskId = tasks[index].status.taskId;
+    tasks.forEach((task) => {
+      const {taskId, runs} = task.status;
+      const runId = runs.length;
```
Comment (djmitche) [inline src/workerinfo.js]: This will start at 1 for runId 0, won't it?

Comment (djmitche) [reply src/workerinfo.js]: (I'm thinking yes given all the runId: 1 in the tests..)

Comment (helfi92) [reply src/workerinfo.js]: Good catch!

**Comment by helfi92** [comment]
@jonasfj I wrote it like the following:

```js
await worker.modify(worker => {
  worker.recentTasks = [
    ...worker.recentTasks,
    ...tasks.map(({status}) => ({taskId: status.taskId, runId: status.runs.length - 1})),
  ].slice(-RECENT_TASKS_LIMIT);
});
```

That way, we make sure it always has the most recent tasks showing in case `tasks` has length greater than `RECENT_TASKS_LIMIT` (assuming this is possible...not sure).

**Pull request merged**